github/compliance/model.py

```diff
@@ -101,14 +107,25 @@ class ScanResultGroup:
     def results_with_successful_scans(self):
         return tuple(r for r in self.results if r.scan_succeeded)
 
+    @property
+    def target(self) -> Target:
+        return self.results[0].scanned_element
+
     @property
     def component(self) -> cm.Component:
-        return self.results[0].scanned_element.component
+        element = self.results[0].scanned_element
+        if not isinstance(element, (cnudie.iter.ResourceNode, cnudie.iter.SourceNode)):
+            raise TypeError(f'unable to get component from {type(element)=}')
```

Comment:
generally speaking, just raise TE w/ offending value as parameter
in this specific case, I would, however, use RuntimeError (since no parameter is explicitly passed here; TE is also okay, I think). the exception's message should state that the used property is not allowed to be used if the scanned_element is not either a RN or a SN (basically repeating the condition above).

if just stating "unable to get component from", this is not very helpful, as it does not explain what the expected condition (in which the property/method would work) are.